node/loop.go

```diff
@@ -177,17 +185,21 @@ func (t *loopingTransport) loop() {
 					delete(t.outboundRequests, msg.ID)
 					t.requestMu.Unlock()
 
-					go func(r *jsonrpc.RawResponse) {
+					go func(o *outboundRequest, r *jsonrpc.RawResponse) {
 						patchedResponse := *r
-						patchedResponse.ID = outbound.request.ID
+						patchedResponse.ID = o.request.ID
 						select {
 						case <-ctx.Done():
 							return
-						case outbound.chResult <- &patchedResponse:
+						case <-o.chAbandoned:
```

Comment:
Does it make sense to need a separate channel here from `ctx.Done()`? Would it be simpler/more semantically consistent if `Request` instead did a `context.WithCancel` before passing it down?

Reply:
The problem then is that we need to store said context in the outboundRequest struct, so that it can be selectable here, and I'm trying to fix an earlier anti-pattern I used where contexts were being stored (which [this page](https://golang.org/pkg/context/) advises against).